subprojects/logging/src/test/groovy/org/gradle/internal/logging/sink/GroupingProgressLogEventGeneratorTest.groovy

```diff
@@ -20,17 +20,31 @@ import org.gradle.api.logging.LogLevel
 import org.gradle.internal.logging.OutputSpecification
```

Comment:
We'll also need test cases that trigger the actual scheduled execution. We might want to think about two things:

- provide a custom time so we don't necessarily have to wait as long as the defined timeout
- make sure that we don't run into timing issues by introducing a count down latch mechanism